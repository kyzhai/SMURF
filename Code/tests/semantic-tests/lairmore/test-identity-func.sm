
ident :: a -> a 
ident x = x
y = ident 1
main = []
