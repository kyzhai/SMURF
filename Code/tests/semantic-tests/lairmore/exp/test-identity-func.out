Program: 
	SFuncdec: 
		ident (x) :: a -> a = x
	New Scope: 
		ID: x  :: a

	
	SVardef: 
		ID: y  :: Int

	ident 1
	SMain: []
Global Scope: 
	ID: main  :: []
	ID: ident x :: a -> a
	ID: y  :: Int
	ID: print x :: a -> a
	ID: random  :: Int
	ID: head h : t :: [a] -> a
	ID: tail h : t :: [a] -> [a]

